packages/utilities/src/lib/codeBlock.ts

```diff
@@ -1,14 +1,27 @@
 const zws = String.fromCharCode(8203);
 
 /**
- * Wraps text in a markdown codeblock with optionally a language indicator for syntax highlighting
+ * Wraps the content inside a codeblock with no language
+ *
+ * @param content - The content to wrap
+ */
+export function codeBlock<C extends string>(content: C): `\`\`\`\n${C}\n\`\`\``;
+
+/**
+ * Wraps the content inside a codeblock with the specified language
+ *
  * @param language The codeblock language
- * @param expression The expression to be wrapped in the codeblock
+ * @param content The expression to be wrapped in the codeblock
  */
-export function codeBlock<T>(language: string, expression: T): string {
-	if (typeof expression === 'string') {
-		if (expression.length === 0) return `\`\`\`${zws}\`\`\``;
-		return `\`\`\`${language}\n${expression.replace(/```/, `\`${zws}\`\``).replace(/`$/g, `\`${zws}`)}\`\`\``;
+export function codeBlock<L extends string, C extends string>(language: L, content: C): `\`\`\`${L}\n${C}\n\`\`\``;
+export function codeBlock(language: string, content?: string): string {
+	if (typeof content === 'string') {
+		if (content.length === 0) {
+			return `\`\`\`\n${zws}\n\`\`\``;
+		}
+
+		return `\`\`\`${language}\n${content.replace(/```/, `\`${zws}\`\``).replace(/`$/g, `\`${zws}`)}\n\`\`\``;
 	}
-	return `\`\`\`${language}\n${expression || zws}\`\`\``;
+
+	return typeof content === 'undefined' ? `\`\`\`\n${language}\n\`\`\`` : `\`\`\`${language}\n${content || zws}\n\`\`\``;
```

Comment:
Is this branch even possible? We already have line 18 for this.

Reply:
If someone in JS writes something like:
```ts
const myCode = someFunc(); // Lets say this returns undefined for whatever reason

codeBlock('js', myCode);
```
then line 18 will be false

So not necessarily

Reply:
That's literally what line 18 does, it already covers the `string` case.

Reply:
`myCode` in my example is undefined so it's NOT covered by the `string` case